### ISSUE TITLE
Using Material UI as Design System

### ISSUE THREAD

**Issue by dyabala** [SoftwareBrothers/admin-bro#536]
Hi, I've discoverd AdminBro recently and it's  really amazing. It has everything I need except I want to customize all the Design System of the UI. Currently AdminBro is using atoms but I'm planning to replace it with Material UI. I know that AdminBro has an option about branding (logo, company name, font,...) but I mean the global styles of all components. So how can I implement this? Thank you.

**Comment by wojtek-krysiak** [comment]
1. regarding the material UI as Design system - it is theoretically possible in v3 (which is in beta right now). We moved entire design-system out to a separate package: @admin-bro/design-system so you can write your own implementation. But it will require to keep all the props and components. Long story short - complicated task, but gradually you can try.
2. You can also clone the repos and do this there.

**Comment by dyabala** [comment]
Thank you for your answer and have a good day. :D

**Comment by perottilds** [comment]
@dyabala  have you manged something out? im giving adminbro a go, but it seems using a material basesd library to not be working

**Comment by wojtek-krysiak** [comment]
do you have any errors? what is happening?

**Comment by perottilds** [comment]
@wojtek-krysiak 

The documentation for the component usage https://copan.loft.com.br/components/button
Here's it being used.
<img width="774" alt="Screen Shot 2020-09-10 at 12 04 03" src="https://user-images.githubusercontent.com/22695166/92750829-d3414680-f35d-11ea-9cae-d5c8d0541fbd.png">

Here's the UI error
<img width="1587" alt="Screen Shot 2020-09-10 at 12 03 54" src="https://user-images.githubusercontent.com/22695166/92750832-d3d9dd00-f35d-11ea-9318-c590d3928631.png">

Console output
<img width="1906" alt="Screen Shot 2020-09-10 at 12 03 26" src="https://user-images.githubusercontent.com/22695166/92750801-cde3fc00-f35d-11ea-890d-9e63998296ed.png">
<img width="1904" alt="Screen Shot 2020-09-10 at 12 03 42" src="https://user-images.githubusercontent.com/22695166/92750837-d50b0a00-f35d-11ea-8792-ac3f06d87c6c.png">

**Comment by perottilds** [comment]
I must add that it's built 100% on top of material-ui. It provides it's own version of material ui, meaning material-ui and its sister projects like styles and labs should be marked as peerDependencies. Maybe that's where it's going a-wire

**Comment by wojtek-krysiak** [comment]
when you initialize AdminBro

const admin = new AdminBro(...)

next run

admin.watch()

it will bundle components in the watch mode meaning there will be more logs. And let me know what happend

**Comment by perottilds** [comment]
btw, I had to alter node_modules/@admin-bro/nestjs/build/admin.module.js directly. Id be nice to have a debug option no it (id glad open a PR for that).
just to be clear: yes im trying it on nestjs
by altering it directly in aforementioned file, I noticed no difference. logs wre the same, no different logs in chrome's console
<img width="1202" alt="Screen Shot 2020-09-11 at 00 55 08" src="https://user-images.githubusercontent.com/22695166/92856216-93b04400-f3c9-11ea-8b22-ae11f84f4aa3.png">

i might try first the simpler one: raw express with my design system
maybe it's something with nestjs, but as far as ive seen, the code looks fine

**Comment by wojtek-krysiak** [comment]
nestjs is super new - we lanched it like 3 days ago so errors are possible, express has more than a year :) and its much more stable

**Comment by perottilds** [comment]
I tried going for express this time. I got the debug mode working.
As you can see I am having issues with bundling.

<img width="1223" alt="Screen Shot 2020-09-12 at 21 58 04" src="https://user-images.githubusercontent.com/22695166/93010056-7eb8e980-f55e-11ea-9ea5-bf30aca5fc9f.png">

I tried updating rollup plugins to the official ones [here](https://github.com/rollup/plugins) but faced some more bundling problems given the apis changed a bit, thus integration between the projects should be redone a bit. As I have little knowledge of the project I went with the imperative strategy: tried to add those named exports directly to rollup config as it was [already being done](https://github.com/mui-org/material-ui/issues/18791#issuecomment-574275944).

Result: same thing.

ps: my design system's rollup config was missing a lib in "external"
ps2: I might have messed up trying to upgrade rollup on both admbrodesignsystem and admbro, I can try again some time
ps3: [this](https://github.com/styled-components/styled-components/issues/1654#issuecomment-684051971) was the motivation to try and update rollup configs

**Comment by perottilds** [comment]
Just to clarify and isolate the issue: going with material-ui directly yields the very same result.
<img width="503" alt="Screen Shot 2020-09-13 at 02 03 50" src="https://user-images.githubusercontent.com/22695166/93010678-6ac4b600-f565-11ea-9c74-7157e27cc2e0.png">
<img width="1419" alt="Screen Shot 2020-09-13 at 02 04 42" src="https://user-images.githubusercontent.com/22695166/93010687-816b0d00-f565-11ea-9569-45aa9715e4cf.png">

**Comment by dyabala** [comment]
@perottilds This is my workarounds:
__*admin-bro/src/backend/bundler/config.js*__
```
Line 10: const reactIsExport = ['isValidElementType', 'isContextConsumer', 'isElement', 'ForwardRef', 'Memo', 'isFragment']
 
```
After that, build this package and use it as private package on my server.

**Comment by wojtek-krysiak** [comment]
problems with bundling external dependencies should be fixed with #614
